src/app/child-dev-project/children/children-count-dashboard/children-count-dashboard.component.ts

```diff
@@ -1,54 +1,84 @@
-import { Component, OnInit } from "@angular/core";
+import { Component, Input, OnInit } from "@angular/core";
 import { ChildrenService } from "../children.service";
 import { Router } from "@angular/router";
-import { UntilDestroy, untilDestroyed } from "@ngneat/until-destroy";
 import { OnInitDynamicComponent } from "../../../core/view/dynamic-components/on-init-dynamic-component.interface";
+import { take } from "rxjs/operators";
+import { ConfigurableEnumValue } from "../../../core/configurable-enum/configurable-enum.interface";
 
-@UntilDestroy()
 @Component({
   selector: "app-children-count-dashboard",
   templateUrl: "./children-count-dashboard.component.html",
   styleUrls: ["./children-count-dashboard.component.scss"],
 })
 export class ChildrenCountDashboardComponent
   implements OnInitDynamicComponent, OnInit {
+  /**
+   * The property of the Child entities to group counts by.
+   *
+   * Default is "center".
+   */
+  @Input() groupBy: string = "center";
+
   totalChildren: number;
-  childrenByCenter = [];
+  childrenGroupCounts = [];
 
   constructor(
     private childrenService: ChildrenService,
     public router: Router
   ) {}
 
-  onInitFromDynamicConfig(config: any) {}
+  onInitFromDynamicConfig(config: any) {
+    if (config?.groupBy && config.groupBy !== this.groupBy) {
```

Comment:
Is there a reason why it needs to be checked if the `groupBy` attribute is the same?

Reply:
originally I had called ngOnInit here and wanted to avoid duplicate db access, this is not relevant here anymore. removing it.